crates/matrix-sdk-crypto/src/store/mod.rs

```diff
@@ -833,16 +833,27 @@ pub(crate) enum UserKeyQueryResult {
 pub struct RoomSettings {
     /// The encryption algorithm that should be used in the room.
     pub algorithm: EventEncryptionAlgorithm,
+
     /// Should untrusted devices receive the room key, or should they be
     /// excluded from the conversation.
     pub only_allow_trusted_devices: bool,
+
+    /// The maximum time an encryption session should be used for, before it is
+    /// rotated.
+    pub session_rotation_period: Option<Duration>,
+
+    /// The maximum number of messages an encryption session should be used for,
+    /// before it is rotated.
+    pub session_rotation_period_msgs: Option<usize>,
```

Comment:
Can we use long names please, like `session_rotation_period_for_messages` maybe?

Reply:
We can, but it might be worth noting that this name is based on the name in the matrix protocol (`rotation_period_msgs`, https://spec.matrix.org/v1.9/client-server-api/#mroomencryption)

I'm not really sure I can think of a better name than `session_rotation_period_messages`. `number_of_messages_to_rotate_session_after` feels very cumbersome.

Reply:
Oh OK. Then `session_rotation_period_messages` is fine for me. Thanks!

Reply:
> We can, but it might be worth noting that this name is based on the name in the matrix protocol (`rotation_period_msgs`, https://spec.matrix.org/v1.9/client-server-api/#mroomencryption)

There is also a `rotation_period_msgs` in `EncryptionSettings`: https://github.com/matrix-org/matrix-rust-sdk/blob/1d6c01fb342507b917697d02392686c861cbd0a7/crates/matrix-sdk-crypto/src/olm/group_sessions/outbound.rs#L82

Reply:
> Oh OK. Then session_rotation_period_messages is fine for me. Thanks! 

Ok!